### ISSUE TITLE
Return the vulnerability scan results if a user attempts to run a container based on a vulnerable image

### ISSUE THREAD

**Issue by kreamyx** [vmware/vic#4568]

**Comment by hickeng** [comment]
@kreamyx I'm not seeing anything in the PRD that asks us to provide this direct feedback.
While we could potentially cram some output into the error message mechanism it's not necessarily going to work well via an API call instead of a docker cli client.
We should see if:
a. we actually need to return details via the docker API or just an error.
b. whether there's scope for adding this information to inspect output instead - that's a much more rich and flexible structure.

My expectation is that what we will be able to return is dictated solely by the response from the registry when we try to pull a vulnerable image.

**Comment by kreamyx** [comment]
@hickeng yes, I think this feature is majorly for Admiral. If the logic of preventing vulnerable images from running is at the Engine side (which seems likely to be the case), then an api request to Harbor to check for vulnerabilities will be necessary. I am inclined more towards returning an appropriate error message that states that the container in question cannot run because of a vulnerable image.

Details in the inspect output is a good idea. But like you said, it is not strictly required as per the PRD
